plugins/hypervisors/kvm/src/main/java/com/cloud/hypervisor/kvm/resource/wrapper/LibvirtGetUnmanagedInstancesCommandWrapper.java

```diff
@@ -0,0 +1,113 @@
+        } catch (Exception e) {
+            s_logger.error("GetUnmanagedInstancesCommand failed due to " + e.getMessage());
+            throw new CloudRuntimeException("GetUnmanagedInstancesCommand failed due to " + e.getMessage());
+        }
+        return new GetUnmanagedInstancesAnswer(command, "True", unmanagedInstances);
+    }
+
+    private UnmanagedInstanceTO getUnmanagedInstance(LibvirtComputingResource libvirtComputingResource, Domain domain) {
+        try {
+            final LibvirtDomainXMLParser parser = new LibvirtDomainXMLParser();
+            parser.parseDomainXML(domain.getXMLDesc(0));
+
+            final UnmanagedInstanceTO instance = new UnmanagedInstanceTO();
+            instance.setName(domain.getName());
+            instance.setCpuCores((int) LibvirtComputingResource.countDomainRunningVcpus(domain));
+            instance.setCpuCoresPerSocket(parser.getCpuModeDef().getCoresPerSocket());
+            instance.setCpuSpeed(parser.getCpuTuneDef().getShares());
+            instance.setMemory((int) LibvirtComputingResource.getDomainMemory(domain));
+
+            // TODO: Ayush complete this function.
+//            instance.setOperatingSystemId(domain.getVmGuestInfo().getGuestId());
+//            if (StringUtils.isEmpty(instance.getOperatingSystemId())) {
+//                instance.setOperatingsSystemId(domain.getConfigSummary().getGuestId());
+//                instance.setOperatingSystemId(domain.getOSType());
+//            }
+//            VirtualMachineGuestOsIdentifier osIdentifier = VirtualMachineGuestOsIdentifier.OTHER_GUEST;
+//            try {
+//                osIdentifier = VirtualMachineGuestOsIdentifier.fromValue(instance.getOperatingSystemId());
+//            } catch (IllegalArgumentException iae) {
+//                if (StringUtils.isNotEmpty(instance.getOperatingSystemId()) && instance.getOperatingSystemId().contains("64")) {
+//                    osIdentifier = VirtualMachineGuestOsIdentifier.OTHER_GUEST_64;
+//                }
+//            }
+//            instance.setOperatingSystem(domain.getGuestInfo().getGuestFullName());
+//            if (StringUtils.isEmpty(instance.getOperatingSystem())) {
+//                instance.setOperatingSystem(domain.getConfigSummary().getGuestFullName());
+//            }
+//            UnmanagedInstanceTO.PowerState powerState = UnmanagedInstanceTO.PowerState.PowerUnknown;
+//            if (domain.getPowerState().toString().equalsIgnoreCase("POWERED_ON")) {
+//                powerState = UnmanagedInstanceTO.PowerState.PowerOn;
+//            }
+//            if (domain.getPowerState().toString().equalsIgnoreCase("POWERED_OFF")) {
+//                powerState = UnmanagedInstanceTO.PowerState.PowerOff;
+//            }
+//            instance.setPowerState(powerState);
+//            instance.setDisks(getUnmanageInstanceDisks(domain));
+//            instance.setNics(getUnmanageInstanceNics(hyperHost, domain));
```

Comment:
@nvazquez - raised this WIP PR for calling out details I am still building for KVM produced UnmanagedInstanceTO.